File: homework3/src/main/java/com/example/homework3/EditContactActivity.java
package com.example.homework3;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
public class EditContactActivity extends AppCompatActivity implements View.OnClickListener {
    private EditText editText1;
    private EditText editText2;
    private Intent intent;
    private Intent resultIntent;
    private boolean isPhone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_contact);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        editText1 = findViewById(R.id.editText1);
        editText2 = findViewById(R.id.editText2);
        intent = getIntent();
        editText1.setText(intent.getStringExtra("name"));
        editText2.setText(intent.getStringExtra("data"));
        findViewById(R.id.buttonArrow).setOnClickListener(this);
        findViewById(R.id.buttonOk).setOnClickListener(this);
        findViewById(R.id.removeButton).setOnClickListener(this);
        resultIntent = new Intent();
        isPhone = intent.getBooleanExtra("isPhone",false);
        if(isPhone) editText2.setHint(R.string.hint2);
        else editText2.setHint(R.string.hint3);
    }

    @Override
    public void onClick(View view) {
        switch(view.getId()){
            case R.id.buttonArrow:
                resultIntent.putExtra("status",0);
                setResult(RESULT_CANCELED,resultIntent);
                finish();
                break;
            case R.id.buttonOk:
                String name = editText1.getText().toString();
                String data = editText2.getText().toString();
                resultIntent.putExtra("status",1);
                resultIntent.putExtra("name", name);
                resultIntent.putExtra("data", data);
                resultIntent.putExtra("isPhone",isPhone);
                resultIntent.putExtra("position",intent.getIntExtra("position",-5));
                setResult(RESULT_OK,resultIntent);
                finish();
            case R.id.removeButton:
                resultIntent.putExtra("status",2);
                resultIntent.putExtra("position",intent.getIntExtra("position",-5));
                setResult(RESULT_OK,resultIntent);
                finish();
        }
    }
}

File: homework6/src/main/java/com/example/homework6/EditContactActivity.kt
package com.example.homework6

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.homework6.db.ContactItem
import com.example.homework6.db.DatabaseManager
import kotlinx.android.synthetic.main.activity_edit_contact.*

class EditContactActivity : AppCompatActivity(), View.OnClickListener {
    private var id : Long = 0
    private lateinit var contact : ContactItem
    private lateinit var databaseManager : DatabaseManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_contact)
        setSupportActionBar(toolbar)
        databaseManager = DatabaseManager.getInstance(applicationContext)
        id = intent.getLongExtra("id",1)
        contact = ContactItem(intent.getStringExtra("name"),intent.getStringExtra("data"),intent.getBooleanExtra("isPhone",false))
        contact.id=id
        editText1.setText(contact.name)
        editText2.setText(contact.data)

        buttonArrow.setOnClickListener(this)
        buttonOk.setOnClickListener(this)
        removeButton.setOnClickListener(this)
        if (contact.isPhone) editText2.setHint(R.string.hint2) else editText2.setHint(R.string.hint3)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.buttonArrow -> {
                finish()
            }
            R.id.buttonOk -> {
                val name = editText1.text.toString()
                val data = editText2.text.toString()
                contact.name = name
                contact.data = data
                databaseManager.updateContact(contact)
                finish()
            }
            R.id.removeButton -> {
                databaseManager.delContactById(id)
                finish()
            }
        }
    }
}